test/src/test-cppapi-ndrectangle.cc

```diff
@@ -77,3 +71,26 @@ TEST_CASE_METHOD(
   CHECK(range[0] == 30);
   CHECK(range[1] == 40);
 }
+
+TEST_CASE("NDRectangle - Errors", "[cppapi][ArraySchema][NDRectangle]") {
+  tiledb::Context ctx;
+
+  // Create a domain
+  tiledb::Domain domain(ctx);
+  auto d1 = tiledb::Dimension::create<int32_t>(ctx, "d1", {{1, 10}}, 5);
+  auto d2 = tiledb::Dimension::create(
+      ctx, "d2", TILEDB_STRING_ASCII, nullptr, nullptr);
+  domain.add_dimension(d1);
+  domain.add_dimension(d2);
+
+  // Create an NDRectangle
+  tiledb::NDRectangle ndrect(ctx, domain);
+
+  // Set range with non-existent dimension
+  CHECK_THROWS(ndrect.set_range(2, 1, 2));
+  CHECK_THROWS(ndrect.set_range("d3", 1, 2));
+
+  // Set range out of order
+  CHECK_THROWS(ndrect.set_range(0, 2, 1));
+  CHECK_THROWS(ndrect.set_range("d2", "bbb", "aaa"));
```

Comment:
I figured out why this fails sometimes. This call gets resolved to the scalar `set_range` overload, where `T` is `const char*`, and passes invalid data to the Core (the address of the pointer variable, and `sizeof(char*)`).

The obvious fix would be to write `std::string("bbb")` in the test, but this masks a defect in the C++ API. My idea of fixing this is:

* Change the string overloads to accept a `string_view` instead of a `string` (we don't need to require allocating a fresh string, since we already pass the length to the C API).
* Add a specialization for the generic scalar overload for `const char*`, which redirects to the `string_view` overload.

c.c. @eric-hughes-tiledb for thoughts.